TrigScint/src/TrigScint/TrigScintFirmwareHitProducer.cxx

```diff
@@ -0,0 +1,97 @@
+
+#include "TrigScint/TrigScintFirmwareHitProducer.h"
+
+#include <iterator>
+#include <map>
+
+#include "TrigScint/Firmware/hitproducer.h"
+#include "TrigScint/Firmware/objdef.h"
+
+namespace trigscint {
+
+void TrigScintFirmwareHitProducer::configure(
+    framework::config::Parameters &ps) {
+  pedestal_ = ps.getParameter<double>("pedestal");
+  gain_ = ps.getParameter<double>("gain");
+  mevPerMip_ = ps.getParameter<double>("mev_per_mip");
+  pePerMip_ = ps.getParameter<double>("pe_per_mip");
+  inputCollection_ = ps.getParameter<std::string>("input_collection");
+  testCollection_ = ps.getParameter<std::string>("test_collection");
+  doTest_ = ps.getParameter<bool>("do_test");
+  inputPassName_ = ps.getParameter<std::string>("input_pass_name");
+  outputCollection_ = ps.getParameter<std::string>("output_collection");
+  verbose_ = ps.getParameter<bool>("verbose");
+  sample_of_interest_ = ps.getParameter<int>("sample_of_interest");
+  if (verbose_) {
+    ldmx_log(info) << "In TrigScintFirmwareHitProducer: configure done!";
+    ldmx_log(info) << "\nPedestal: " << pedestal_ << "\nGain: " << gain_
+                   << "\nMEV per MIP: " << mevPerMip_
+                   << "\nPE per MIP: " << pePerMip_
+                   << "\ninput collection:     " << inputCollection_
+                   << "\ntest collection:	" << testCollection_
+                   << "\nAre we testing:        " << doTest_
+                   << "\nInput pass name:     " << inputPassName_
+                   << "\nOutput collection:    " << outputCollection_
+                   << "\nVerbosity: " << verbose_;
+  }
+
+  return;
+}
+
+void TrigScintFirmwareHitProducer::produce(framework::Event &event) {
+  if (doTest_) {
+    const auto rechits{event.getCollection<ldmx::TrigScintHit>(testCollection_,
+                                                               inputPassName_)};
+    for (const auto &hit : rechits) {
+      std::cout << "Analysis barID: " << hit.getBarID()
+                << ", PE Number: " << hit.getPE() << std::endl;
+    }
+  }
+  const auto digis{event.getCollection<trigscint::TrigScintQIEDigis>(
+      inputCollection_, inputPassName_)};
+  Hit outHit[NHITS];
+  ap_uint<14> FIFO[NCHAN][5];
+  ap_uint<8> Peds[NCHAN];
+  for (int i = 0; i < NCHAN; i++) {
+    Peds[i] = 0;
+    FIFO[i][0] = (Peds[i] << 6) + 63;
+    FIFO[i][1] = (Peds[i] << 6) + 63;
+    FIFO[i][2] = (Peds[i] << 6) + 63;
+    FIFO[i][3] = (Peds[i] << 6) + 63;
+    FIFO[i][4] = (Peds[i] << 6) + 63;
+  }
+  std::cout << "DID I GET HERE 3" << std::endl;
+  for (const auto &digi : digis) {
+    std::vector<int> adcs = digi.getADC();
+    std::vector<int> tdcs = digi.getTDC();
+    for (int i = 0; i < 5; i++) {
```

Comment:
is the 5 the anyhow connected to the `NTIMES` variable?

Reply:
Yes. The number 5 used to be wholely determined by the clock speed of the FPGA and our bunch crossing rate. Namely if we are able to have a ~5 ns clock cycle with ~25-26 many clocks between bunches, than we have a latency of 5. Originally this latency was what was going to restrict how many clockcycles we integrated a pulse over, but now we just need it as long as a pulse as we'll run a couple of these in parrallel. This number, however, will not change and is somewhat nonconfigurable; the firmware is built around it. I would need to delete lines to make it 3 or add some to make it 6.